### ISSUE TITLE
SFT val/dir_acc plateaus at ~91% — looks like irreducible label noise

### ISSUE THREAD

**Issue by beyarkay** [beyarkay/factorion#106]
## Symptom

The first SFT sweep ([wandb](https://wandb.ai/beyarkay/factorion/sweeps/ncqdnvmt), PR #104) saturated at `val/acc ≈ 0.90` across the top 7 runs. Per-head breakdown of the rank-1 run reveals the direction head is the sole bottleneck:

| Head | val acc |
|---|---|
| tile | 0.984 |
| ent  | 0.992 |
| **dir** | **0.912** |
| item | 1.000 |
| misc | 0.997 |
| eot  | 0.998 |

All top 5 runs are essentially identical *except* for ~1pp of variance on `val/dir_acc`. Bayes effectively optimized "which run got the luckiest dir head."

## Per-lesson-kind breakdown (top run)

`dir` is the bottleneck on every kind:

| kind                | acc  | tile | ent  | dir  | item | misc |
|---------------------|------|------|------|------|------|------|
| MOVE_ONE_ITEM       | 0.81 | 0.93 | 1.00 | 0.86 | 1.00 | 1.00 |
| ASSEMBLE_1IN_1OUT   | 0.85 | 0.98 | 0.97 | 0.88 | 1.00 | 1.00 |
| MOVE_VIA_UG_BELT    | 0.88 | 0.97 | 0.99 | 0.91 | 1.00 | 0.98 |
| INSERTER_TRANSFER   | 0.93 | 1.00 | 1.00 | 0.93 | 1.00 | 1.00 |
| SPLITTER_SPLIT      | 0.93 | 1.00 | 1.00 | 0.93 | 1.00 | 1.00 |
| SPLITTER_MERGE      | 0.93 | 1.00 | 1.00 | 0.93 | 1.00 | 1.00 |

## This is not a compute or capacity ceiling

The training curve of the top run:

```
epoch   train_acc   val_acc   val/dir_acc   val_loss
 5        0.84        0.84       0.87         0.74
10        0.89        0.88       0.89         0.44
15        0.91        0.89       0.90         0.40
20        0.92        0.90       0.91         0.39
25        0.93        0.90       0.91         0.38–0.42 (bouncing)
30        0.93        0.90       0.91         0.40
```

`train_acc` keeps climbing while val plateaus; `val_loss` minimum is around epoch 17 and drifts up after. More data / more epochs / bigger encoder won't fix this — the dir head has run out of signal.

## Hypothesis: irreducible aleatoric uncertainty from lesson-generator symmetry

At early-step intermediate states (mostly-blank world), direction is genuinely ambiguous from the observation alone. A 3-belt path that's still mostly blank could canonically run N→S, S→N, E→W, or W→E and every direction is equally consistent with the partial state. The lesson generator picks one at random per seed, but the model has no way to know which choice was made.

If true, this isn't a model bug — it's a dataset bug. The labels are noisier than the inputs can support.

## Suggested investigations

1. **Confirm the hypothesis by logging `val/dir_acc` by step index.** Modify `extract_expert_actions` to emit a `step_index` (0 = mostly blank, N = nearly full) and bucket val/dir_acc by it. The hypothesis predicts step 0 is much worse than step N.
2. **Canonicalize ambiguous configurations in the lesson generator.** When multiple directions are equally valid completions, the generator should always pick the same one (e.g., lexicographic-smallest path).
3. **Or: provide more context to the model.** A "what entities have been placed in previous steps" channel would break symmetry on early-step states.

## Why this matters for the SFT → RL pipeline

If the SFT ceiling is structurally 0.91 on direction, the downstream PPO policy starts with a 9-percentage-point hole on every direction-bearing action. Either the SFT signal needs cleaning (option 1/2 above) or RL has to learn to disambiguate from rewards — both have real cost.

Found during analysis of PR #104 sweep results.
